### PR TITLE
Support creation from file or json object (#8)

### PR DESCRIPTION
**Pull request by cfstout**

### CODE REVIEW

File: json_regex_diff/jsondiff.py [hunk 1]
```diff
@@ -7,36 +7,60 @@
...
+
+        try:
+            new_json = json.load(open(json_file))
+        except IOError:
+            logger.error("JSON File not found. Check name and try again")
+            new_json = None
             exit(1)
 
-    def _set_up_model_map(self, json_model):
+        is_directory = not os.path.isfile(json_model)
```
Comment (derek-rushing) [inline json_regex_diff/jsondiff.py]: Why not os.path.isdir here? If there is a specific reason probably add a comment.

File: json_regex_diff/jsondiff.py [hunk 2]
```diff
@@ -7,36 +7,60 @@
...
+
+        #Set up model map
         model_map = {}
         if os.path.isfile(json_model):
             try:
                 model_map[json_model] = json.load(open(json_model))
             except IOError:
-                self._logger.error("Model file not found. Check name and try again")
+                logger.error("Model file not found. "
```
Comment (derek-rushing) [inline json_regex_diff/jsondiff.py]: self._logger

File: json_regex_diff/jsondiff.py [hunk 3]
```diff
@@ -46,19 +70,27 @@ def _set_up_model_map(self, json_model):
                     filename = json_model + item
                     model_map[item] = json.load(open(filename))
                 except IOError:
-                    self._logger.error("Could not open file")
+                    logger.error("Could not open file")
```
Comment (derek-rushing) [inline json_regex_diff/jsondiff.py]: self._logger

File: json_regex_diff/jsondiff.py [hunk 4]
```diff
@@ -46,19 +70,27 @@ def _set_up_model_map(self, json_model):
...
         else:
-            self._logger.error("File or directory not found."
-                         " Check name and try again.")
+            logger.error("File or directory not found. "
```
Comment (derek-rushing) [inline json_regex_diff/jsondiff.py]: self._logger

File: json_regex_diff/jsondiff.py [hunk 5]
```diff
@@ -7,36 +7,60 @@
...
+        :param json_model: Path to old json file or directory of models to
+        compare with
+        ** Important to note that multiple 'models' are not currently supported
+        when computing a diff **
+        """
+
+        try:
+            new_json = json.load(open(json_file))
+        except IOError:
+            logger.error("JSON File not found. Check name and try again")
```
Comment (derek-rushing) [inline json_regex_diff/jsondiff.py]: self._logger

File: json_regex_diff/jsondiff.py [hunk 6]
```diff
@@ -46,19 +70,27 @@ def _set_up_model_map(self, json_model):
...
-            self._logger.error("File or directory not found."
-                         " Check name and try again.")
+            logger.error("File or directory not found. "
+                               "Check name and try again.")
             exit(1)
-        return model_map
 
-    def _clear_match_row(self, match_table, row, cur_index):
+        if len(model_map) < 1:
+            logger.error("No files could be read in specified directory")
```
Comment (derek-rushing) [inline json_regex_diff/jsondiff.py]: self._logger

Comment (derek-rushing) [inline json_regex_diff/jsondiff.py]: self._logger

File: json_regex_diff/jsondiff.py (hunk 2 above)
Comment (cfstout) [inline json_regex_diff/jsondiff.py]: Moved inside a class method, so no reference to self available.

File: json_regex_diff/jsondiff.py (hunk 3 above)
Comment (cfstout) [inline json_regex_diff/jsondiff.py]: Moved inside a class method, so no reference to self available.

File: json_regex_diff/jsondiff.py (hunk 4 above)
Comment (cfstout) [inline json_regex_diff/jsondiff.py]: Moved inside a class method, so no reference to self available.

**Pull request merged**